### DISCUSSION TITLE
ssr-check 运行报错

### DISCUSSION THREAD

**Discussion by zengming00** [immortalwrt/immortalwrt#530] [Q&A] [answered]
```
root@ImmortalWrt:~# ssr-check 
Segmentation fault
```
主机名 | ImmortalWrt
型号 | D-LINK DIR-825 C1
架构 | Atheros AR9344 rev 1
固件版本 | ImmortalWrt 21.02-SNAPSHOT r19047-20552f7d57 / LuCI openwrt-21.02 branch git-21.343.56137-c5039ed
内核版本 | 5.4.158
本地时间 | 2021-12-12 16:27:14

**Comment by zengming00** [comment]
今天刚刚编译的，怀疑是不是浮点运算的问题？编译的时候没加软浮点？

**Comment by zxlhhyccc** [comment]
> 今天刚刚编译的，怀疑是不是浮点运算的问题？编译的时候没加软浮点？

me too。。。。

**Comment by 1715173329** [comment, marked as answer]
我建议你先去学习下 ssr-check 是怎么用的。

**Reply by hecheng337** [reply]
解释一下。
ssr-check不是给人用的，
它的作用是连接一个指定的主机名和端口，把连接状态通过返回值的形式报告给脚本，脚本以此判断代理工作是否正常；
luci-app-ssr-plus利用它来实现 自动切换 功能和 “状态” 页面那个 “检查连通性” 按钮。
（详见 /usr/bin/ssr-switch 和 /usr/lib/lua/luci/controller/shadowsocksr.lua）
直接执行ssr-check会报错，而且就算传入参数、执行成功了，也不会有任何标准输出。

**Reply by chobitsos** [reply]
> 解释一下。 ssr-check不是给人用的， 它的作用是连接一个指定的主机名和端口，把连接状态通过返回值的形式报告给脚本，脚本以此判断代理工作是否正常； luci-app-ssr-plus利用它来实现 自动切换 功能和 “状态” 页面那个 “检查连通性” 按钮。 （详见 /usr/bin/ssr-switch 和 /usr/lib/lua/luci/controller/shadowsocksr.lua） 直接执行ssr-check会报错，而且就算传入参数、执行成功了，也不会有任何标准输出。

正确用法
脚本帮你写好了，保存在/etc/下面记得给755的权限，然后跑一下/etc/脚本名字

#!/bin/sh

/usr/bin/ssr-check www.google.com 80
if [ "$?" == "0" ]; then
    echo "Proxy Success."
else
    echo "Proxy Failed."
fi
